src/basicfuns.jl

```diff
@@ -44,6 +44,31 @@ function xlog1py(x::Number, y::Number)
     return iszero(x) && !isnan(y) ? zero(result) : result
 end
 
+"""
+$(SIGNATURES)
+
+Return `x * exp(x)` for `x > -Inf`, handling `x = -Inf` by taking the downward limit.
+
+```jldoctest
+julia> xexpx(-Inf)
+0.0
+```
+"""
+xexpx(x::Real) = isfinite(x) ? x * exp(x) : exp(x)
+
+"""
+$(SIGNATURES)
+
+Return `x * exp(y)` for `y > -Inf`, handling `y = -Inf` by taking the downward limit.
+
```

Comment:
I find this a bit confusing (how the limit is defined). I think the manual should just say that it is `x * exp(y)` except when ..., and enumerate the special cases and their evaluation rules explicitly.

Reply:
Yes that's better I updated the docstring. Thanks.

Reply:
Perfect.